diff --git a/modules/proxy/mod_proxy_balancer.c b/modules/proxy/mod_proxy_balancer.c
index c59f5e973f..77fa9d6c39 100644
--- a/modules/proxy/mod_proxy_balancer.c
+++ b/modules/proxy/mod_proxy_balancer.c
@@ -24,6 +24,7 @@
 #include "apr_date.h"
 #include "util_md5.h"
 #include "mod_watchdog.h"
+#include <apr_thread_mutex.h>
 
 static const char *balancer_mutex_type = "proxy-balancer-shm";
 ap_slotmem_provider_t *storage = NULL;
diff --git a/modules/proxy/proxy_util.c b/modules/proxy/proxy_util.c
index 6501c68064..b3341a7b47 100644
--- a/modules/proxy/proxy_util.c
+++ b/modules/proxy/proxy_util.c
@@ -14,6 +14,9 @@
  * limitations under the License.
  */
 
+#include <apr_thread_mutex.h>
+#define APR_THREAD_MUTEX_DEFAULT  0x0
+
 /* Utility routines for Apache proxy */
 #include "mod_proxy.h"
 #include "ap_mpm.h"
